dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/NotificationRuleActionImplementer.java

```diff
@@ -120,18 +124,21 @@ protected ProgramNotificationInstance createNotificationInstance( ProgramNotific
         return notificationInstance;
     }
 
-    protected boolean validate( RuleEffect ruleEffect, ProgramInstance programInstance )
+    @Transactional
+    public boolean validate( RuleEffect ruleEffect, ProgramInstance programInstance )
     {
-        if ( ruleEffect == null )
+        if ( ruleEffect == null || programInstance == null )
```

Comment:
Might it be better to cast a np here and use checkNotNull, since a silent fail will mask a potential bug?

Reply:
fixed

Reply:
Is it fixed in the branch and pushed?

Reply:
fixed in the branch but have not pushed it yet.